### DISCUSSION TITLE
CustomDefinitionProvider on void Methods

### DISCUSSION THREAD

**Discussion by PintOfBitter** [victools/jsonschema-generator#440] [Q&A] [answered]
Hello,

I have a class with a very basic void method:
```
    public void voidMethod() {
        String hello = "hello";

        System.out.println(hello);
    }
```

I am trying to implement a custom logic to the generated schema using a CustomPropertyDefinitionProvider:
```
            CustomPropertyDefinitionProvider provider = (member, context) -> {
                _//Do custom logic_
            };
builder.forMethods().withCustomDefinitionProvider(provider);
```

While this works great for non void methods, it seems that void methods are not included with the forMethods() configuration.

The options im using are: 
`without(Option.FLATTENED_SUPPLIERS, Option.FLATTENED_OPTIONALS)`
and
`with(Option.VOID_METHODS, Option.STATIC_METHODS, Option.NONSTATIC_NONVOID_NONGETTER_METHODS)`
With the PLAIN_JSON preset

The resulting schema does contain the void methods but without the customDefinition:
`"voidMethod()" : false`

Is there something I am missing here? Any clue on how to proceed from here is greatly appreciated as I have been on this for a while now.

Thanks,
PoB

**Comment by CarstenWickner** [comment, marked as answer]
Hi @PintOfBitter,

I can confirm your observation and can reproduce it.
The reason for that behavior is this short-circuiting logic for `void` methods at the moment:
https://github.com/victools/jsonschema-generator/blob/f226ad937de9d75d059a11748e6e7b74f6f585b2/jsonschema-generator/src/main/java/com/github/victools/jsonschema/generator/impl/SchemaGenerationContextImpl.java#L593-L596

Any custom definition would only be considered after this check.
Out of curiosity: what kind of schema do you intend to assign to such a `void` method's return value?

An extra caveat could be added to enable at least such a `CustomPropertyDefinition` to be respected.

----

EDIT: I've made the requisite change in #441 and plan to release it as v4.35.0.

**Reply by PintOfBitter** [reply]
Hey @CarstenWickner,

Thanks again for the swift response.
I intend to document fields and methods with certain annotations that will be used in a small generic framework im working on. These annotations will help determine the order of the execution of said methods.
A bit unorthodox but seems to help tremendously with performance in certain cases as I am trying to avoid loading bulky jars.

Thank you for the quick change! Is there a release date planned for v.4.35.0?

Thanks,
PoB

**Reply by CarstenWickner** [reply]
I'll do the release tonight and let you know.

**Reply by CarstenWickner** [reply]
@PintOfBitter it's done: [v4.35.0](https://github.com/victools/jsonschema-generator/releases/tag/v4.35.0)
Should be propagated and available within a couple of hours.
